src/main/java/io/neonbee/internal/deploy/DeployableModels.java

```diff
@@ -0,0 +1,105 @@
+    public static final String NEONBEE_MODELS = "NeonBee-Models";
+
+    /**
+     * The JAR file MANIFEST.MF descriptor for the deployable model extensions / associated models.
+     */
+    public static final String NEONBEE_MODEL_EXTENSIONS = "NeonBee-Model-Extensions";
+
+    @VisibleForTesting
+    final EntityModelDefinition modelDefinition;
+
+    /**
+     * Creates a {@link DeployableModels} by scanning the JAR file for model files to deploy.
+     *
+     * @param vertx   the Vert.x instance used to scan the class path
+     * @param jarPath the path to a JAR file to find the models in
+     * @return a {@link DeployableModels}
+     */
+    public static Future<DeployableModels> fromJar(Vertx vertx, Path jarPath) {
+        return ClassPathScanner.forJarFile(vertx, jarPath).compose(classPathScanner -> {
+            return scanClassPath(vertx, classPathScanner).eventually(classPathScanner.close(vertx));
+        });
+    }
+
+    @VisibleForTesting
+    static Future<DeployableModels> scanClassPath(Vertx vertx, ClassPathScanner classPathScanner) {
+        // to load the models, we can use the same class-loader that the class-path-scanner uses, because we anyways
+        // just use it to read the resources
+        ClassLoader classLoader = classPathScanner.getClassLoader(); // NOPMD false positive getClassLoader
+        Future<Map<String, byte[]>> csnModelDefinitions = classPathScanner.scanManifestFiles(vertx, NEONBEE_MODELS)
+                .compose(csnModelNames -> readModelPayloads(vertx, classLoader, csnModelNames));
+        Future<Map<String, byte[]>> associatedModelDefinitions =
+                classPathScanner.scanManifestFiles(vertx, NEONBEE_MODEL_EXTENSIONS)
+                        .compose(associatedModelNames -> readModelPayloads(vertx, classLoader, associatedModelNames));
+        return CompositeFuture.all(csnModelDefinitions, associatedModelDefinitions)
+                .map(compositeResult -> new EntityModelDefinition(csnModelDefinitions.result(),
+                        associatedModelDefinitions.result()))
+                .map(DeployableModels::new);
+    }
+
+    /**
+     * Creates a {@link DeployableModels} from a {@link EntityModelDefinition}.
+     *
+     * @param modelDefinition the {@link EntityModelDefinition} to deploy
+     */
+    public DeployableModels(EntityModelDefinition modelDefinition) {
+        super();
+        this.modelDefinition = modelDefinition;
+    }
+
+    @Override
+    public String getIdentifier() {
+        return modelDefinition.toString();
+    }
+
+    @Override
+    public PendingDeployment deploy(NeonBee neonBee) {
+        return new PendingDeployment(neonBee, this,
+                neonBee.getModelManager().registerModels(modelDefinition).mapEmpty()) {
+            @Override
+            protected Future<Void> undeploy(String deploymentId) {
+                return neonBee.getModelManager().unregisterModels(modelDefinition).mapEmpty();
+            }
+        };
```

Comment:
Mhm this is a good thing to watch out for, I see what you mean.

In this exact case, I don't think it'll be a problem though, because we anyways store a reference to the deployable the deployment came from as a field in the Deployment and in this case this happens to be passed into the constructor as "this", thus garbage collection would not touch it nevertheless :) So we should be fine here!